.gitignore

```diff
@@ -46,3 +46,13 @@ credentials.json
 
 # tf lock file
 .terraform.lock.hcl
+
+**/terraform.tfstate*
+**/.terraform*
+**/backend.tf
```

Comment:
I'm not sure if we want to add any `*.tf` to gitignore?

CC: @g-awmalik @bharathkkb